target/cheri-common/cheri-lazy-capregs-types.h

```diff
@@ -65,11 +65,24 @@ typedef enum CapRegState {
     /// The tag bit can be read from the cap_register_t structure.
     CREG_FULLY_DECOMPRESSED = 0b11
 } CapRegState;
+
+// Cap registers should be padded so they are easier to move.
+#if TARGET_LONG_BITS == 32
+_Static_assert(sizeof(cap_register_t) == 40, "");
+#else
+_Static_assert(sizeof(cap_register_t) == 64, "");
+#endif
+// pesbt should come directly before reg._cr_cursor, so that the two can be
+// moved with a single 128bit vector op.
+_Static_assert((offsetof(cap_register_t, cached_pesbt) -
+                offsetof(cap_register_t, _cr_cursor)) == sizeof(target_ulong),
+               "");
+
 typedef struct GPCapRegs {
-    target_ulong pesbt[32]; // permissions+exponent+sealing type+bottom+top
+    // We use one of the padding fields as pesbt (to avoid constantly
+    // decompressing values such as $csp which are used frequently).
+    // 33 allows us to have an actual 0 register that is none of the others.
+    cap_register_t decompressed[33];
```

Comment:
I don't like the 33. MIPS and RISC-V have their zero register as one of the 32 already. I assume this is so both (the current EL's) CSP and CZR can live here? This should become a per-target define.

Reply:
Can drop from this one, will need changing later anyway.